streams/src/main/java/org/apache/kafka/streams/KafkaStreams.java

```diff
@@ -351,6 +354,7 @@ public void setUncaughtExceptionHandler(final Thread.UncaughtExceptionHandler eh
      * @param partitioner       Partitioner for the store
      * @param <K>               key type
      * @return  The {@link StreamsMetadata} for the storeName and key
+     * @throws StreamsException if streams is (re-)initializing
```

Comment:
Actually for `store()` call in line 376, we could also throw an `InvalidStateStoreException` from `QueryableStoreProvider.getStore()`, right?